pkgs/development/python-modules/onnx/default.nix

```
@@ -39,6 +39,7 @@ buildPythonPackage rec {
 
   build-system = [
     cmake
+    protobuf_21
     pybind11
```

As it isn't a Python package, `protobuf_21` should be in `nativeBuildInputs` instead.
Also, this list should follow https://github.com/onnx/onnx/blob/v1.16.2/pyproject.toml#L4 or you should open an upstream issue to fix their `pyproject.toml`.
```suggestion
    cmake
    protobuf
    setuptools
```
And please use `pyproject = true`.